### PR TITLE
Updated to work with funcgen ResultSets. 

### PR DESCRIPTION
**Pull request by swingingsimian**
Faked DataFile objects. Added --schema_type option
Tested using --dry_run.

### CODE REVIEW

**Comment by magaliruffier** [comment]
the changes look sane to me

File: scripts/datafiles/build_datafile_ftp_directory.pl [hunk 1]
```diff
@@ -222,16 +277,18 @@ sub _target_species_root {
...
 }
 
+
 # Expected path: base/data_files/normalpath
 # e.g. pub/release-66/data_files/pan_trogladytes/CHIMP2.14/rnaseq/chimp_1.bam
 sub _target_datafiles_root {
   my ($self, $datafile) = @_;
   my $base = File::Spec->catdir($self->opts()->{ftp_dir}, 'data_files');
-  my $target_location = $datafile->path($base);
+  (my $target_location = $datafile->path($base)) =~ s/funcgen\///;
```
Comment (nerdstrike) [inline scripts/datafiles/build_datafile_ftp_directory.pl]: I would prefer this to be two lines, as legibility is poor, and I can't guarantee Perl's behaviour. The brackets could mean list context as much as it controls operator precedence. I'm sure it works, but it makes me cautious.

File: scripts/datafiles/build_datafile_ftp_directory.pl [hunk 2]
```diff
@@ -284,15 +352,25 @@ sub _get_core_like_dbs {
...
+
+  return \%final_dbas;
 }
 
+
 sub _files {
   my ($self, $datafile) = @_;
-  my $source_file = $datafile->path($self->opts()->{datafile_dir});
+
+  (my $source_file = $datafile->path($self->opts()->{datafile_dir})) =~ s/funcgen\///;
```
Comment (nerdstrike) [inline scripts/datafiles/build_datafile_ftp_directory.pl]: As above for line 286.

**Comment by swingingsimian** [comment]
Now handles remaning Monocytes-CD14+ files by using quotemeta.

**Comment by magaliruffier** [comment]
any chance you could split the lines Kieron mentioned before we pull the changes in?

**Comment by thomasmaurel** [comment]
Hello, 

Any idea why Travis didn't pick up the missing ";" line 379? the script couldn't compile because of that.

Cheers,
Thomas

**Comment by nerdstrike** [comment]
Because Travis runs the test suite, not a full check of all scripts. I suppose we could add a test that does actually check syntax on everything with a .pl extension.

**Pull request merged**